packages/internal-test-helpers/lib/get-all-property-names.ts

```diff
@@ -1,9 +1,10 @@
-export default function getAllPropertyNames(Klass: Function) {
+export default function getAllPropertyNames<T>(Klass: { prototype: T }): Set<keyof T & string> {
   let proto = Klass.prototype;
-  let properties: Set<string> = new Set();
+  let properties: Set<keyof T & string> = new Set();
 
   while (proto !== Object.prototype) {
-    let names = Object.getOwnPropertyNames(proto);
+    // SAFETY: This should be safe.
```

Comment:
…but why? :joy: Again, the explanation that the intersection with string gets us string-only properties of `T` is probably enough.